perfschema/perfschema.go

```diff
@@ -33,40 +30,45 @@ type StatementInstrument interface {
 
 // PerfSchema defines the methods to be invoked by the executor
 type PerfSchema interface {
-	// For SELECT statement only.
-	NewPerfSchemaPlan(tableName string) (plan.Plan, error)
-
-	// For INSERT statement only.
-	ExecInsert(insertVals *InsertValues) error
 
 	// For statement instrumentation only.
 	StatementInstrument
+
+	// GetDBMeta returns db info for PerformanceSchema.
+	GetDBMeta() *model.DBInfo
+	// GetTable returns table instance for name.
+	GetTable(name string) (table.Table, bool)
 }
 
 type perfSchema struct {
-	tables map[string]*model.TableInfo
-	fields map[string][]*field.ResultField
-	stores map[string]*leveldb.DB
-	lsns   map[string]*uint64
+	store   kv.Storage
+	dbInfo  *model.DBInfo
+	tables  map[string]*model.TableInfo
+	mTables map[string]table.Table // MemoryTables for perfSchema
+
+	// Used for TableStmtsHistory
+	historyHandles []int64
```

Comment:
There is a little problem about connID. If you access TiDB from tidb-server, there will be a connID. If you access TiDB from sql-driver, the connID will always be zero.